server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -696,6 +696,10 @@ private static void failIfMatchesRoutingPath(DocumentParserContext context, Stri
      */
     private static void parseCopyFields(DocumentParserContext context, List<String> copyToFields) throws IOException {
         for (String field : copyToFields) {
+            if (context.mappingLookup().getMapper(field) instanceof InferenceFieldMapper) {
+                // ignore copy_to that targets inference fields, values are already extracted in the coordinating node to perform inference.
+                continue;
+            }
```

Comment:
drive-by comment: could we try and avoid this type of instanceof checks / special cases? What would it take to remove it? Would it be worthwhile to generalize the special needs of inference field mapper into Mapper somehow? Other ideas? I think it's important that we at least consider it.